src/metrics/representation.py

```diff
@@ -146,228 +150,225 @@ def compute(self) -> float:
+            sort_indexes = np.argsort(current_scores)
+            relevant_idxs = relevant_idxs[sort_indexes[::-1]]
+            relevants.append(relevant_idxs)
         
-        return q_vecs, db_vecs, relevant, scores, db_idxs, queries_idxs
+        relevants = np.array(relevants)
+        return relevants, clear_scores, gallery_idxs, queries_idxs, clear_query_order_numbers
 
-    def prepare_classification_data(self, vectors: np.ndarray, targets: np.ndarray
-    ) -> Tuple[np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray, np.ndarray]:
+    def prepare_classification_data(self, 
+                                    targets: np.ndarray) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
         """Prepare data for faiss build index, and following search, in classification dataset case.
         
-        Separate query and database vectors from storage vectors.
-        Query vector index will be the first row index for each unique target value found in targets array. 
-        So, total number of queries is equal to unique target count. 
-        The remaining vectors will be treated as database vectors (again for each unique target value)
+        In the classification case, ll the vectors would be used as queries, and the relevants vectors will be vectors 
+        that have the same label as the query.
 
         Args:
-            vectors: Vectors of the whole storage, includes queries and database vectors, 
-                size (dataset_size, embedding_size).
-            targets: Targets in classification task for every vector, size (database_size).
-
-        Retruns:
-            q_vecs: Queries vectors, size (queries_size, embedding_size).
-            db_vecs: Database vectors, for faiss build index, size (database_size, embedding_size).
-            relevant: Array of relevant indexes in database for every query request, size (queries_size, ).
-            scores: Array of scores without queries empty scores.
-            db_idxs: Array with all database indexes.
-            queries_idxs: Array of queries order number.
+            targets: Targets in classification task for every vector.
+
+        Returns:
+            relevants: Array of relevant indexes in gallery data, for every query.
+            gallery_idxs: Array of gallery indexes in vectors storage.
+            queries_idxs: Array of queries indexes in vectors storage.
+
+        Raises:
+            ValueError: If any class has only one element.
         """
         ts = pd.Series(targets)
-
-        # Group item indices of the same class
         groups = pd.Series(ts.groupby(ts).groups)
+        # now groups contain group indexes with one targets
+        relevants = []
+        queries_idxs = []
+        # create queries with its relevants indexes
+        for i, group in enumerate(groups):
+            for query_idx in group:
+                # need to drop from relevants index which equal query index
```

Comment:
Why did you change the implementation back to pandas version? It was good IMHO

Reply:
it was bad implementation, because for `shuffle=True` it won't be work. It was only works when class arranged in order like [0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3 ....]. If not use pandas, we need to create groups like in this implementation, so we need call np.uniq then np.where len(np.uniq) times and only then create cycle for delete query element.